cmd/cli/uninstall_mesh.go

```diff
@@ -80,70 +81,303 @@ func newUninstallMeshCmd(config *action.Configuration, in io.Reader, out io.Writ
+			fmt.Fprintf(d.out, "No OSM control plane with mesh name [%s] found in namespace [%s]\n", d.meshName, d.meshNamespace)
 
-		confirm, err := confirm(d.in, d.out, fmt.Sprintf("\nUninstall OSM [mesh name: %s] in namespace [%s] ?", d.meshName, ns), 3)
-		if !confirm || err != nil {
-			return err
+			if !d.deleteClusterWideResources && !d.deleteNamespace {
+				return err
+			}
 		}
-	}
 
-	_, err := d.client.Run(d.meshName)
-	if err != nil && errors.Cause(err) == helmStorage.ErrReleaseNotFound {
-		if d.force {
-			return nil
+		if err == nil {
+			fmt.Fprintf(d.out, "OSM [mesh name: %s] in namespace [%s] uninstalled\n", d.meshName, d.meshNamespace)
 		}
-		return errors.Errorf("No OSM control plane with mesh name [%s] found in namespace [%s]", d.meshName, ns)
+	} else {
+		fmt.Fprintf(d.out, "OSM [mesh name: %s] in namespace [%s] CANNOT be uninstalled in a managed environment\n", d.meshName, d.meshNamespace)
 	}
 
-	if err == nil {
-		fmt.Fprintf(d.out, "OSM [mesh name: %s] in namespace [%s] uninstalled\n", d.meshName, ns)
+	if d.deleteClusterWideResources {
+		var failedDeletions []string
+
+		err = d.uninstallCustomResourceDefinitions()
+		if err != nil {
+			failedDeletions = append(failedDeletions, "CustomResourceDefinitions")
+		}
+
+		err = d.uninstallMutatingWebhookConfigurations()
+		if err != nil {
+			failedDeletions = append(failedDeletions, "MutatingWebhookConfigurations")
+		}
+
+		err = d.uninstallValidatingWebhookConfigurations()
+		if err != nil {
+			failedDeletions = append(failedDeletions, "ValidatingWebhookConfigurations")
+		}
+
+		err = d.uninstallSecrets()
+		if err != nil {
+			failedDeletions = append(failedDeletions, "Secrets")
+		}
+
+		if len(failedDeletions) != 0 {
+			return errors.Errorf("Failed to completely delete the following OSM resource types: %+v", failedDeletions)
+		}
 	}
 
 	if d.deleteNamespace {
-		if err = d.clientSet.CoreV1().Namespaces().Delete(ctx, ns, v1.DeleteOptions{}); err != nil {
-			return errors.Errorf("Error occurred while deleting OSM namespace [%s] - %v", ns, err)
+		if !settings.IsManaged() {
+			if err = d.clientSet.CoreV1().Namespaces().Delete(ctx, d.meshNamespace, v1.DeleteOptions{}); err != nil {
+				if k8sApiErrors.IsNotFound(err) {
+					fmt.Fprintf(d.out, "OSM namespace [%s] not found\n", d.meshNamespace)
+					return nil
+				}
+				return errors.Errorf("Error occurred while deleting OSM namespace [%s] - %v", d.meshNamespace, err)
+			}
+			fmt.Fprintf(d.out, "OSM namespace [%s] deleted successfully\n", d.meshNamespace)
+		} else {
+			fmt.Fprintf(d.out, "OSM namespace [%s] CANNOT be deleted in a managed environment\n", d.meshNamespace)
 		}
-		fmt.Fprintf(d.out, "OSM namespace [%s] deleted successfully\n", ns)
 	}
 
-	return err
+	return nil
+}
+
+// validateCLIParams contains all checks necessary to ensure that various permutations of the CLI flags are consistent
+func (d *uninstallMeshCmd) validateCLIParams() error {
```

Comment:
Since there are defaults for these flags, why is this validation necessary?

Reply:
good catch, it's not necessary removed it